source/mqtt/Mqtt5Client.cpp

```diff
@@ -233,6 +247,36 @@ namespace Aws
 
             Mqtt5ClientOptions::~Mqtt5ClientOptions() {}
 
+            Mqtt5to3AdapterOptions *Mqtt5ClientOptions::NewMqtt5to3AdapterOptions() const noexcept
+            {
+                Mqtt5to3AdapterOptions *adapterOptions = new Mqtt5to3AdapterOptions();
+                adapterOptions->m_hostName = m_hostName;
+                adapterOptions->m_port = m_port;
+                adapterOptions->m_socketOptions = m_socketOptions;
+                if (m_proxyOptions.has_value())
+                    adapterOptions->m_proxyOptions = m_proxyOptions.value();
+                if (m_tlsConnectionOptions.has_value())
+                {
+                    adapterOptions->m_tlsConnectionOptions = m_tlsConnectionOptions.value();
+                }
+                if (websocketHandshakeTransform)
+                {
+                    adapterOptions->m_overwriteWebsocket = true;
+
+                    auto signerTransform = [this](
+                                               std::shared_ptr<Crt::Http::HttpRequest> req,
+                                               const Crt::Mqtt::OnWebSocketHandshakeInterceptComplete &onComplete) {
+                        this->websocketHandshakeTransform(req, onComplete);
```

Comment:
`websocketHandshakeTransform` takes its first parameter by value, so `req` should be moved here